seml/database.py

```diff
@@ -277,3 +281,40 @@ def clean_unreferenced_artifacts(db_collection_name=None, yes=False):
     logging.info('Deleting not referenced artifacts...')
     delete_files(db, not_referenced_artifacts, progress=True)
     logging.info(f'Successfully deleted {n_delete} not referenced artifact{s_if(n_delete)}.')
+
+def list_database(pattern, mongodb_config=None, progress=False, list_empty=False):
+    """
+    Prints a tabular version of multiple collections and their states (without resolving RUNNING experiments that may have been canceled manually).
+
+    Parameters
+    ----------
+    pattern : str
+        The regex collection names have to match against
+    mongodb_config : dict or None
+        A configuration for the mongodb. If None, the standard config is used.
+    progress : bool
+        Whether to use a progress bar for fetching
+    list_empty : bool
+        Whether to list collections that have no documents associated with any state
+    """
+    logging.warning(f"Status of {States.RUNNING[0]} experiments may not reflect if they have died or been canceled. Use `seml ... status` instead.")
+    if mongodb_config is None:
+        mongodb_config = get_mongodb_config()
+    db = get_database(**mongodb_config)
+    collection_names = [name for name in db.list_collection_names()
+                        if name not in ('fs.chunks', 'fs.files') and re.compile(pattern).match(name)]
+    name_to_counts = {}
+    it = tqdm(collection_names) if progress else collection_names
+    
+    for collection_name in it:
+        counts_by_status = db[collection_name].aggregate([{'$group' : {'_id' : '$status', '_count' : {'$sum' : 1}}}])
+        name_to_counts[collection_name] = Counter({result['_id'] : result['_count'] for result in counts_by_status})
+    
+    columns = [States.STAGED, States.PENDING, States.RUNNING, States.FAILED, States.KILLED, States.INTERRUPTED, 
+               States.COMPLETED]
+    table = []
+    for name, counts in name_to_counts.items():
+        if list_empty or any(sum(counts[state] for state in states) > 0 for states in columns):
```

Comment:
Okay, this comment is now roughly 30min later, and a bunch of unnecessary testing later. I did not super like the long if statement, the reiteration of the list, and the misuse of a counter. So, I thought: can't we just directly use a dataframe. And the answer is yes:
```python
    logging.warning(f"Status of {States.RUNNING[0]} experiments may not reflect if they have died or been canceled. Use `seml ... status` instead.")
    if mongodb_config is None:
        mongodb_config = get_mongodb_config()
    db = get_database(**mongodb_config)
    collection_names = [name for name in db.list_collection_names()
                        if name not in ('fs.chunks', 'fs.files') and re.compile(pattern).match(name)]
    name_to_counts = defaultdict(lambda: {state: 0 for state in States.keys()})
    it = tqdm(collection_names) if progress else collection_names
    
    inv_states = {v: k for k, states in States.items() for v in states}
    for collection_name in it:
        counts_by_status = db[collection_name].aggregate([{'$group' : {'_id' : '$status', '_count' : {'$sum' : 1}}}])
        name_to_counts[collection_name].update({
            inv_states[result['_id']]: result['_count']
            for result in counts_by_status
            if result['_id'] in inv_states
        })
    
    df = pd.DataFrame.from_dict(name_to_counts, dtype=int).transpose()
    # Remove empty collections
    if not list_empty:
        df = df[df.sum(axis=1) > 0]
    # add a column with the total
    df['Total'] = df.sum(axis=1)
    logging.info(df.sort_index().to_string())
```
This is purely to please my own ego and to shorten a long line of codes and I definitely don't want to diminish your work!

Reply:
Personally, I think this is less readable, which, for non-runtime critical applications like seml, I prefer to be the maxime. However, if you want this version, I have no objections, I only care about the feature being realized.

Reply:
Interesting, I thought of it being more readable ^^ but that may be because I authored it or because of the comments. But this implementation also just takes the ordering from settings.py which may be desired(?).

Reply:
As I said, take yours if you prefer it, the ordering from settings.py is definitely a good point that one can and should realize.